client/blocks/upe/fields.js

```diff
@@ -63,6 +89,127 @@ const UPEField = ( {
+			paymentMethodsConfig.card !== undefined
+		) {
+			const shippingAddressFields = {
+				line1: 'shipping-address_1',
+				line2: 'shipping-address_2',
+				city: 'shipping-city',
+				state: 'components-form-token-input-1',
+				postal_code: 'shipping-postcode',
+				country: 'components-form-token-input-0',
+			};
+			const billingAddressFields = {
+				line1: 'billing-address_1',
+				line2: 'billing-address_2',
+				city: 'billing-city',
+				state: 'components-form-token-input-3',
+				postal_code: 'billing-postcode',
+				country: 'components-form-token-input-2',
+			};
+
+			const appearance = getAppearance();
+			elements.update( { appearance } );
+
+			enableStripeLinkPaymentMethod( {
+				api,
+				elements,
+				emailId: 'email',
+				fill_field_method: ( address, nodeId, key ) => {
+					const setAddress =
+						shippingAddressFields[ key ] === nodeId
+							? customerData.setShippingAddress
+							: customerData.setBillingData;
+					const customerAddress =
+						shippingAddressFields[ key ] === nodeId
+							? customerData.shippingAddress
+							: customerData.billingData;
+
+					if ( undefined === customerAddress ) {
+						return;
+					}
+
+					if ( address.address[ key ] === null ) {
+						address.address[ key ] = '';
+					}
+
+					if ( key === 'line1' ) {
+						customerAddress.address_1 = address.address[ key ];
+					} else if ( key === 'line2' ) {
+						customerAddress.address_2 = address.address[ key ];
+					} else if ( key === 'postal_code' ) {
+						customerAddress.postcode = address.address[ key ];
+					} else {
+						customerAddress[ key ] = address.address[ key ];
+					}
+
+					setAddress( customerAddress );
+
+					function getEmail() {
+						return document.getElementById( 'email' ).value;
+					}
+
+					if ( undefined !== customerData.billingData ) {
+						customerData.billingData.email = getEmail();
+						customerData.setBillingData( customerData.billingData );
+					}
+				},
+				show_button: ( linkAutofill ) => {
+					jQuery( '#email' )
+						.parent()
+						.append(
+							'<button class="stripe-gateway-stripelink-modal-trigger"></button>'
+						);
+					if ( jQuery( '#email' ).val() !== '' ) {
+						jQuery(
+							'.stripe-gateway-stripelink-modal-trigger'
+						).show();
+
+						const linkButtonTop =
+							jQuery( '#email' ).position().top +
+							( jQuery( '#email' ).outerHeight() - 40 ) / 2;
+						jQuery(
+							'.stripe-gateway-stripelink-modal-trigger'
+						).show();
+						jQuery(
+							'.stripe-gateway-stripelink-modal-trigger'
+						).css( 'top', linkButtonTop + 'px' );
+					}
+
+					//Handle StripeLink button click.
+					jQuery( '.stripe-gateway-stripelink-modal-trigger' ).on(
+						'click',
+						( event ) => {
+							event.preventDefault();
+							// Trigger modal.
+							linkAutofill.launch( {
+								email: jQuery( '#email' ).val(),
+							} );
+						}
+					);
+				},
+				complete_shipping: () => {
```

Comment:
I think that after latest change the `complete_shipping` is now a function, but [here](https://github.com/woocommerce/woocommerce-gateway-stripe/blob/86248da286bd08c0447a6d0d22f75d530cb48b73/client/stripe-link/index.js#L51) we are checking if the property exists, instead of calling this function. Note that this check will always return true:

`if ( options.complete_shipping ) {` 

As far as I understand, it should be:

`if ( options.complete_shipping() ) {` 

Can you take a look please?